lib/primer/classify.rb

```diff
@@ -202,11 +202,13 @@ def extract_value(memo, key, val, breakpoint)
         elsif key == COLOR_KEY
           char_code = val[-1].ord
           # Does this string end in a character that is NOT a number?
-          memo[:classes] << if char_code >= 48 && char_code <= 57 # 48 is the charcode for 0; 57 is the charcode for 9
-                              "color-#{val.to_s.dasherize}"
-                            else
-                              "text-#{val.to_s.dasherize}"
-                            end
+          memo[:classes] <<
+            if (char_code >= 48 && char_code <= 57) || # 48 is the charcode for 0; 57 is the charcode for 9
+               val =~ /(primary|secondary|tertiary|link|success|warning|danger|info)/
```

Comment:
It looks like this is causing the allocations test to fail since this is introducing an allocation, I think we can fix it by moving `/(primary|secondary|tertiary|link|success|warning|danger|info)/` to a constant, like `COLOR_REGEX` and changing this line to:

```suggestion
               COLOR_REGEX.match?(val)
```

Thoughts?

Reply:
Thanks, that did the trick, though the same builds are now failing with something I don't think my change introduced?
```
activesupport-7.0.0.alpha requires ruby version >= 2.7.0, which is incompatible
with the current version, ruby 2.6.6p146
Error: Process completed with exit code 5.
```

Reply:
Hmmm, I'm not sure why that suddenly started happening. I can take a look shortly.

Reply:
Looks like it resolved when Joel merged in main!

Reply:
Ah yeah, https://github.com/primer/view_components/pull/194 got merged which fixed the build :)